### PR TITLE
Use only forward slashes for paths during build

### LINKED ISSUE #628: Always use forward slash for file paths when generating the website
**Tell us about your environment**

* **MarkBind Version:** 1.17.0

**What did you do? Please include the actual source code causing the issue.**

Repeat the steps on 2 different platforms (Windows vs Mac/Linux):
1. `markbind init`
2. `markbind build`
3. Check the file paths inside the generated `index.html`.

**What did you expect to happen?**

The file paths generated should use a consistent slash (i.e. `/`), regardless of the OS that the author is running.

**What actually happened? Please include the actual, raw output.**

* On Windows, the slash generated is `\`.

```html
<script src="markbind\js\setup.js"></script>
<script src="markbind\layouts\default\scripts.js"></script>
</html>
```

* On Mac & Linux, the slash generated is `/`.

```html
<script src="markbind/js/setup.js"></script>
<script src="markbind/layouts/default/scripts.js"></script>
</html>
```

### PR DESCRIPTION
**Pull request by Xenonym**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Enhancement to an existing feature

Resolves #628.

**What is the rationale for this request?**
When MarkBind generates paths during site build, it uses the default path separator of the OS it is running on (`\` on Windows, `/` on Mac/Linux). For consistency, MarkBind should use the same path seperator regardless of OS.

**What changes did you make? (Give an overview)**
I created a temporary `assets` object which has all the paths run through `ensurePosix` to build the template with all forward slashes its paths. I also modified the parser to generate paths with forward slashes for dynamic `panel` includes.

**Testing instructions:**
1. `markbind build` a site. The generated HTML should use forward slashes regardless of OS.

### CODE REVIEW

File: src/Page.js [hunk 1]
```diff
@@ -214,8 +214,18 @@ Page.prototype.prepareTemplateData = function () {
...
     : this.title;
 
+  // construct temporary asset object with only POSIX-style paths
+  const asset = {};
+  Object.entries(this.asset).forEach(([key, value]) => {
+    if (key === 'externalScripts') {
+      asset[key] = value;
+    } else {
+      asset[key] = FsUtil.ensurePosix(value);
+    }
```
Comment (nicholaschuayunzhi) [inline src/Page.js]: We could shorten this to:
```js
asset[key] = key === 'externalScripts' ? value : FsUtil.ensurePosix(value);
```

File: src/Page.js [hunk 2]
```diff
@@ -214,8 +214,18 @@ Page.prototype.prepareTemplateData = function () {
     ? this.titlePrefix + (this.title ? TITLE_PREFIX_SEPARATOR + this.title : '')
     : this.title;
 
+  // construct temporary asset object with only POSIX-style paths
+  const asset = {};
+  Object.entries(this.asset).forEach(([key, value]) => {
+    if (key === 'externalScripts') {
```
Comment (nicholaschuayunzhi) [inline src/Page.js]: Since `ensurePosix` requires a `string`, perhaps we could use `typeof`

**Comment by Xenonym** [comment]
> Have you considered writing a wrapper over `path`. While i understand it may hurt build time but it could potentially result in less developer error.

While it would be possible to have our own `path` that always uses forward slashes, I am not sure if it will create interoperability issues with other libaries, since they will most likely be relying on the default `path`with the platform specific `path.sep`. Passing paths between other libraires and our own code in this case would probably lead to unexpected results.

File: src/Page.js (hunk 2 above)
Comment (Xenonym) [reply src/Page.js]: As recommended by [`eslint-plugin-lodash/prefer-lodash-typecheck`](https://github.com/wix/eslint-plugin-lodash/blob/master/docs/rules/prefer-lodash-typecheck.md), I have opted for `_.isString(value)`instead.

**Comment by nicholaschuayunzhi** [comment]
@Xenonym Thanks for the changes, please rebase on `master`, thanks!

**Comment by yamgent** [comment]
As discussed on Slack the `enablePosix()` thing is duplicated across two different util scripts. Perhaps the one in `src/lib/markbind/src/utils.js` can stay, while methods that are calling the `src/util/fsUtil.js` version can switch to `src/lib/markbind/src/utils.js`?

**Comment by Xenonym** [comment]
@yamgent consolidated `ensurePosix` uses to the one from `src/lib/markbind/src/utils.js`!
